gen/src/main.rs

```diff
@@ -0,0 +1,130 @@
+extern crate getopts;
+
+extern crate futures;
+extern crate hyper;
+extern crate tokio_core;
+
+#[macro_use]
+extern crate serde_derive;
+extern crate serde;
+extern crate serde_yaml;
+
+#[macro_use]
+extern crate lazy_static;
+
+#[macro_use]
+extern crate matches;
+extern crate regex;
+
+mod download;
+mod generate;
+
+use std::collections::HashMap;
+use std::env;
+use std::fs::File;
+use std::io::BufReader;
+use std::path::Path;
+
+use download::download;
+use generate::generate_all;
+
+use getopts::Options;
```

Comment:
👌 I chose `getopts` fairly arbirarily. Since I'm going to do a minor rewrite of the harness while switching to TOML, I'll go ahead and make the switch over.